lib/bundles/inspec-compliance/target.rb

```diff
@@ -13,7 +13,48 @@ module Compliance
   class Fetcher < Fetchers::Url
     name 'compliance'
     priority 500
-    def self.resolve(target) # rubocop:disable PerceivedComplexity, Metrics/CyclomaticComplexity, Metrics/AbcSize
+    attr_reader :upstream_sha256
+
+    def initialize(target, opts)
+      super(target, opts)
+      if target.is_a?(Hash) && target.key?(:url)
+        @target = target[:url]
+        @upstream_sha256 = target[:sha256]
+      elsif target.is_a?(String)
+        @target = target
+        @upstream_sha256 = ''
+      end
+    end
+
+    def sha256
+      upstream_sha256.empty? ? super : upstream_sha256
+    end
+
+    def self.check_compliance_token(config)
+      if config['token'].nil? && config['refresh_token'].nil?
+        if config['server_type'] == 'automate'
+          server = 'automate'
+          msg = 'inspec compliance login https://your_automate_server --user USER --ent ENT --dctoken DCTOKEN or --token USERTOKEN'
+        elsif config['server_type'] == 'automate2'
```

Comment:
Will this work with all version of Automate 2 or just the latest one?

Reply:
Yes, it should work with all versions of Automate 2. The logic for check_compliance_token is unchanged. I took it out of the resolve method because Metrics/AbcSize pointed to the resolve method having too many lines. Since it was a discrete piece of logic, I broke it out into its own method. You can compare it to the old code here: https://github.com/inspec/inspec/pull/3221/files#diff-e975c4d3a237ff3f9a0f22cc54abab4aL32